### PR TITLE
#1091 ruff added to pre-commit, pyproject.toml, requirements-dev.txt, ruff linting differences 

### PR DESCRIPTION
**Pull request by salice**

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -107,19 +107,18 @@ encountered. If no failures are identified, then you are good to go!
 **Note:*** Some tests are written to call out to external API’s, and will be skipped as part of standard unit testing.
 This is expected.
 
-See the [pytest documentation](https://docs.pytest.org/en/latest/contents.html) for more info and many more options.
```
Comment (austinweisgrau) [inline CONTRIBUTING.md:110]: This link should not have been changed

File: requirements-dev.txt [hunk 2]
```diff
@@ -1,11 +1,30 @@
 # Testing Requirements
```
Comment (austinweisgrau) [inline requirements-dev.txt:1]: Are all of these additions really necessary? Couldn't we just add ruff?

**Review by austinweisgrau — CHANGES_REQUESTED** [review]

File: CONTRIBUTING.md (hunk 1 above)
Comment (salice) [reply CONTRIBUTING.md]: File is updated with current version

**Review by salice — COMMENTED** [review]
CONTRIBUTING.md file is consistent on branch

File: requirements-dev.txt (hunk 2 above)
Comment (salice) [reply requirements-dev.txt:1]: Updated and removed other packages

**Review by austinweisgrau — APPROVED** [review]

**Pull request merged**